### DISCUSSION TITLE
tixiGetNumberOfChilds gives 1 for an empty list

### DISCUSSION THREAD

**Discussion by spookyGh0st** [DLR-SC/tixi#231] [Announcements]
For the xml file:
```xml
<?xml version="1.0" encoding="utf-8"?>
<project>
  <elements>
  </elements>
</project>
```
And the corresponding function call:
```cpp
int count;
tixiGetNumberOfChilds(handle,"project/elements",&count);
```
we get 1 and not as I would expect to 0. Is this intentional, and if so, how would one get the actual list size?

**Comment by ArneBachmannDLR** [comment]
Confirmed for Python:
```python3
> set PATH=C:\Users\bach_ar\Downloads\TIXI-3.3.0-win64\bin;%PATH%
> python
import tixi3wrapper
cpacs = tixi3wrapper.Tixi3()
cpacs.openString("""?xml version="1.0" encoding="utf-8"?>
<project>
  <elements>
  </elements>
</project>""")
cpacs.getNumberOfChilds("/project/elements")
1
```

I can fix it by removing the newline and spaces between `<element>` and `</element>`.
It seems that tixi counts the space as an element somehow.

**Comment by rainman110** [comment]
Spaces and new Lines constitute technically to (invisible) XML text elements, so that's why you'll get "1" as the number of childs. So technically (by xml definition), the result 1 is correct. I understand though, that in this case, you want probably iterate over the "element" nodes, so you expect a zero count.

I think a better API would be something

```c
tixiGetNumberOfChildElements(handle, "/project/elements", "element", &count)
```
    
where you explicitly need to defined the element name you are searching for.

You can though use the function `tixiXPathEvaluateNodeNumber` instead, to query only for "element" nodes:

```c
tixiXPathEvaluateNodeNumber(handle, "/project/elements//element", &count)
```

**Comment by rainman110** [comment]
Here's an example

```python
from tixi3 import tixi3wrapper

cpacs = tixi3wrapper.Tixi3()
cpacs.openString("""<?xml version="1.0" encoding="utf-8"?>
<project>
  <elements>
  </elements>
</project>""")

# prints 0
print(cpacs.xPathEvaluateNodeNumber("/project/elements//element"))

cpacs = tixi3wrapper.Tixi3()
cpacs.openString("""<?xml version="1.0" encoding="utf-8"?>
<project>
  <elements>
    <element/>
  </elements>
</project>""")

# prints 1
print(cpacs.xPathEvaluateNodeNumber("/project/elements//element"))
```

**Comment by CLiersch** [comment]
I guess, this would be equivalent to the
_**ReturnCode tixiGetNamedChildrenCount (const TixiDocumentHandle handle, const char *elementPath, const char *childName, int *count)**_
method, right?

**Comment by rainman110** [comment]
@CLiersch You are absolutely right 🤦 . It seems, that it is too long ago, since I was using tixi last time.

**Comment by rainman110** [comment]
I am closing the issue, since the function is doing what it should do. Use instead `tixiGetNamedChildrenCount` as @CLiersch suggested.

**Comment by joergbrech** [comment]
Thanks for your input! I moved this issue to the discussion board.
